modules/luci-mod-dashboard/htdocs/luci-static/resources/view/dashboard/include/10_router.js

```diff
@@ -0,0 +1,322 @@
+				this.params.internet.v4.gatewayv4.value =  ifc.getGatewayAddr() || '0.0.0.0';
+				this.params.internet.v4.connected.value = ifc.isUp();
+				this.params.internet.v4.addrsv4.value = ifc.getIPAddrs() || [ '-'];
+				this.params.internet.v4.dnsv4.value = ifc.getDNSAddrs() || [ '-' ];
+			 }
+		}
+	},
+
+	renderInternetBox: function(data) {
+
+		this.params.internet = {
+
+			v4: {
+				title: 'IPv4 Internet',
+
+				connected: {
+					title: _('Connected'),
+					visible: true,
+					value: false
+				},
+
+				uptime: {
+					title: _('Connected since'),
+					visible: true,
+					value: '-'
+				},
+
+				protocol: {
+					title: _('Protocol'),
+					visible: true,
+					value: '-'
+				},
+
+				addrsv4: {
+					title: _('IPv4'),
+					visible: true,
+					value: [ '-' ]
+				},
+
+				gatewayv4: {
+					title: _('GatewayV4'),
+					visible: true,
+					value: '-'
+				},
+
+				dnsv4: {
+					title: _('DNSv4'),
+					visible: true,
+					value: ['-']
+				}
+			},
+
+			v6: {
+				title: 'IPv6 Internet',
+
+				connected: {
+					title: _('Connected'),
+					visible: true,
+					value: false
+				},
+
+				uptime: {
+					title: _('Connected since'),
+					visible: true,
+					value: '-'
+				},
+
+				protocol: {
+					title: _('Protocol'),
+					visible: true,
+					value: ' - '
+				},
+
+				ipprefixv6 : {
+					title: _('IPv6 prefix'),
+					visible: true,
+					value: ' - '
+				},
+
+				addrsv6: {
+					title: _('IPv6'),
+					visible: false,
+					value: [ '-' ]
+				},
+
+				gatewayv6: {
+					title: _('GatewayV6'),
+					visible: true,
+					value: '-'
+				},
+
+				dnsv6: {
+					title: _('DNSv6'),
+					visible: true,
+					value: [ '-' ]
+				}
+			}
+		};
+
+		this.renderUpdateWanData(data[0], false);
+		this.renderUpdateWanData(data[1], true);
+
+		return this.renderHtml(this.params.internet, 'internet');
+	},
+
+	renderRouterBox: function(data) {
+
+		var boardinfo   = data[2],
+			systeminfo  = data[3];
+
+		var datestr = null;
+
+		if (systeminfo.localtime) {
+			var date = new Date(systeminfo.localtime * 1000);
+
+			datestr = '%04d-%02d-%02d %02d:%02d:%02d'.format(
+				date.getUTCFullYear(),
+				date.getUTCMonth() + 1,
+				date.getUTCDate(),
+				date.getUTCHours(),
+				date.getUTCMinutes(),
+				date.getUTCSeconds()
+			);
+		}
+
+		this.params.router = {
+			uptime: {
+				title: _('Uptime'),
+				value: systeminfo.uptime ? '%t'.format(systeminfo.uptime) : null,
+			},
+
+			localtime: {
+				title: _('Local Time'),
+				value: datestr
+			},
+
+			kernel: {
+				title: _('Kernel Version'),
+				value: boardinfo.kernel
+			},
+
+			model: {
+				title: _('Model'),
+				value: boardinfo.model
+			},
+
+			system: {
+				title: _('Architecture'),
+				value: boardinfo.system
+			},
+
+			release: {
+				title: _('Firmware Version'),
+				value: (L.isObject(boardinfo.release) ? boardinfo.release.version + ' ('+boardinfo.release.revision+')' : '')
```

Comment:
The test if `boardinfo.release` is an object doesn't work here. Confusingly it's always a release.

Here is a SNAPSHOT vs a 19.07.2 `ubus call system board` output:
```
{
	"kernel": "5.4.52",
	"hostname": "gw",
	"system": "ARMv7 Processor rev 1 (v7l)",
	"model": "Linksys WRT3200ACM",
	"board_name": "linksys,wrt3200acm",
	"release": {
		"distribution": "OpenWrt",
		"version": "SNAPSHOT",
		"revision": "r13936+7-739d282c2f",
		"target": "mvebu/cortexa9",
		"description": "OpenWrt SNAPSHOT r13936+7-739d282c2f"
	}
}
```
vs
```
{
	"kernel": "4.14.171",
	"hostname": "core1.reudnetz.org",
	"system": "AMD G-T40E Processor",
	"model": "PC Engines APU",
	"board_name": "pc-engines-apu",
	"release": {
		"distribution": "OpenWrt",
		"version": "19.07.2",
		"revision": "r10947-65030d81f3",
		"target": "x86/64",
		"description": "OpenWrt 19.07.2 r10947-65030d81f3"
	}
}
```

Either you check if `version == "SNAPSHOT"` or you just always use `description`.